addons/account/models/chart_template.py

```diff
@@ -257,6 +258,38 @@ def _load(self, sale_tax_rate, purchase_tax_rate, company):
                     res.with_context(force_delete=True).unlink()
             existing_accounts.unlink()
 
+        # Propagate set fields from ancestor to current coa
+        fields_to_prop = {
+            "currency_id",
+            "country_id",
+            'bank_account_code_prefix',
+            'cash_account_code_prefix',
+            'transfer_account_code_prefix',
+        }
+        update_vals = {"code_digits": self._default_code_digits}
+
+        for chart in self.search([('id', 'parent_of', self.id), ('id', '!=', self.id)]):
+            for field in fields_to_prop:
+                update_vals[field] = getattr(chart, field) or update_vals.get(field, False)
+
+            if chart.code_digits and chart.code_digits != self._default_code_digits:
+                update_vals["code_digits"] = getattr(chart, "code_digits")
+
+        self.write(update_vals)
+
+        # Check if required fields that were unset in this coa has been found in ancestors
+        fields_to_prop.remove("country_id")
+        erroneous_fields = set()
+        for field_name in fields_to_prop:
+            field_value = getattr(self, field_name, None)
+            missing_currency = field_name == "currency_id" and field_value.id is None
+            if not field_value or missing_currency:
+                erroneous_fields.add(field_name)
+        if erroneous_fields:
+            raise ValidationError(
+                    f"Missing required field(s) data in parent hierarchy for {self.name}: {erroneous_fields}"
+            )
+
```

Comment:
You would just write on the company in a loop
(pseudo code)
```python
for coa in self.parent_ids:
    company.write({
        ...
    })
```

Don't change current record's values on the CoA

Reply:
When does it loop? _load is only called once on the last template being loaded when you install a template. Am I missing something?

Reply:
> Am I missing something?

This: 
> Don't change current record's values on the CoA

This method is not expected to change anything on the CoA, but only to load it on the company.